packages/fxa-react/extract-imported-components.js

```diff
@@ -0,0 +1,45 @@
+const { readdirSync, readFileSync, statSync } = require('fs');
+const { resolve } = require('path');
+
+const FXA_REACT_IMPORT_REGEX = /import(?:["'\s]*[\w*{}\n\r\t, ]+from\s*)?["'\s].*(fxa-react\/components\/[\w\/.]+)["'\s].*$/gm;
+
+function eligibleFile(file) {
+  return (
+    ['.jsx', '.tsx', '.js', '.ts'].some((ext) => file.endsWith(ext)) &&
+    !file.includes('.test')
+  );
+}
+
+function getExtractableFiles(dir) {
+  const subdirs = readdirSync(dir);
+  return subdirs
+    .map((subdir) => {
+      const res = resolve(dir, subdir);
+      return statSync(res).isDirectory() ? getExtractableFiles(res) : res;
+    })
+    .reduce((arr, files) => arr.concat(files), [])
```

Comment:
for some code golf try [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)